### DISCUSSION TITLE
Midi CC print to oled

### DISCUSSION THREAD

**Discussion by masydoblig** [tttapa/Control-Surface#974] [Q&A]
Hi 

I have finally come back to this project and implemented the motor fader it works beautifully with cubase and the generic remote. Its fast and accurate. 
When i get time i will post a you tube video of the build.

I Have also managed to make a display interface for the ST7735 colour tft which works well with the Mackie control.
Great work so far Peter it really is a good project.

My last piece of the puzzle is to get the track names using the generic remote function.

Is there a way to print the incoming midi cc to the OLED.
I have been trying to get the track names out of Cubase and it works with the MCU implementation, but i want to use the generic controller.

I need to make a display element like the MCU LCDDisplay 
Or copy it but when i do i get an error that MCU does not name a type.

Im not sure if anyone has made a display element similar to this yet.

What would i need to include in it to use the assignment display input element?

any help would be appreciated @tttapa 

Cheers

**Comment by tttapa** [comment]
> I have been trying to get the track names out of Cubase and it works with the MCU implementation, but i want to use the generic controller.

Are you sure that Cubase actually sends the track names when using a generic controller? What data format does it use to encode the text and the track numbers? What type of MIDI messages does it use?

**Comment by masydoblig** [comment]
Here is the out put from cubase using the mackie reverse engineering code, this is going from one channel to another


![image](https://github.com/tttapa/Control-Surface/assets/39230113/560e7893-866f-45e1-9715-1dad7aa8142e)

I can set it to send out the name of the track in the generic editor as below and when i change channels it sends the message.
ignore the midi mixer one (it also sends the name data and i was using it to check if they are the same)
![20231113_193511](https://github.com/tttapa/Control-Surface/assets/39230113/680e96f1-9efa-4461-a2fe-07d79675f4d4)

![20231113_193612](https://github.com/tttapa/Control-Surface/assets/39230113/6a0152c3-6674-474f-bfc4-e2671d480323)

**Reply by tttapa** [reply]
> Here is the out put from cubase using the mackie reverse engineering code, this is going from one channel to another

That does indeed look like a sequence of NRPN messages, but there is no text data in it. It simply selects address 64 (`b0 63 00`, `b0 62 40`), and then it sets the 14-bit data to the value 16383 (`b0 06 7f`, `b0 26 7f`), but this value does not seem meaningful to me, it's just the maximum value that NRPN supports. I would expect some kind of ASCII-encoded multi-byte message.

**Reply by masydoblig** [reply]
Hi Peter

yeah you are right.

If you ever come across a way to get them out let me know.

Other than that the controller works great for me.

Thanks for the help
